src/components/MultiSelectGroup/MultiSelectGroup.mdx

```diff
@@ -0,0 +1,125 @@
+- {<Link href="#passing-multiselect-children" target="_self">Passing `MultiSelect` Children</Link>}
+- {<Link href="#changelog" target="_self">Changelog</Link>}
+
+## Overview
+
+<Description of={MultiSelectGroupStories} />
+
+## Component Props
+
+<Canvas of={MultiSelectGroupStories.WithControls} />
+
+<ArgTypes of={MultiSelectGroupStories.WithControls} />
+
+## Accessibility
+
+The `MultiSelectGroup` renders a group of `MultiSelect` components that are
+wrapped in a `<fieldset>` element. The `<fieldset>` element renders a `<legend>`
+element that can be visually hidden through the `showLabel` prop. If the
+`<legend>` is hidden an `aria-label` attribute with `labelText` value will be
+rendered instead.
+
+For accessibility information on the `MultiSelect` component, see the
+[MultiSelect Accessibility](../?path=/docs/components-form-elements-multiselect--docs#accessibility) documentation.
+
+## Layout Patterns
+
+The width of the MultiSelect components can be passed through the
+`multiSelectWidth` prop, with values of `"fitContent"` or `"full"`.
+On screen width under 600px width the `layout` is automatically set to "column"
+and the `multiSelectWidth` to "full". Note: when `layout = "column"` is passed,
+the MultiSelect components will have `isBlockElement` set to `true` by default.
+
+**IMPORTANT**: we do not recommended setting `multiSelectWidth` to `"full"` and
+`layout` to `"row"` at the same time. When the `MultiSelect` components are
+opened, the overlap will cause z-index issues with the selected count button.
+
+<Canvas of={MultiSelectGroupStories.LayoutPatterns} />
+
+## useMultiSelect hook
+
+`useMultiSelect` is an optional but recommended hook for cases where managing
+the state of the component in the consuming app is less of a concern and general
+ease of use is prefered.
+
+The hook returns an object containing all the props and state needed to handle
+the selectedItems. That includes the functions `onChange`, `onClearAll`,
+`onMixedStateChange` for handling any changes to the selection of items and the
+current state of the selection: `selectedItems`. Additionally it returns
+`setSelectedItems` for setting an initial state of `selectedItems` or manipulate
+the `selectedItems` if needed.
+
+Due to its data structure and functions setup, it is capable to handle a group
+of `MultiSelect` components. Find the full documentation under
+[useMultiSelect](../?path=/docs/hooks-usemultiselect--docs).
+
+## Passing MultiSelect Children
+
+In order to optimize rendering `MultiSelect` children and control the
+`isBlockElement` and `multiSelectWidth` props, the `MultiSelectGroup` component
+accepts `MultiSelect` children through its render prop `renderMultiSelect`
+function. This callback function returns an object argument with two optional
+properties: `isBlockElement` and `multiSelectWidth`.
+
+Both `isBlockElement` and `multiSelectWidth` are passed to the `MultiSelect`
+children.
```

Comment:
Do we need this: "In order to optimize rendering `MultiSelect` children and control the
`isBlockElement` and `multiSelectWidth` props,"? I already find this paragraph a bit confusing, and I think this adds to the confusion without explaining how it optimizes rendering or controls the props.

Reply:
I think it's needed. It's not explicit why it's optimizing the rendering but it's doing so because the `react.cloneElement` function was removed internally. I do not want to flag that in the docs because I don't think it's useful to explain/write about internal code. I can reword this.